Create the skeleton structure for the app

jbasden opened the pull request:
This commit creates the metric_publisher directory and the associated files.
app.py shows up as green because I added it back in. Other than that, there has been no code added. I just created the folders and files needed.

Commit:
Create the skeleton structure for the app
This commit creates the metric_publisher directory and the associated files.

jlhood reviewed (changes requested):
You have *.pyc files in your changes. Those should never be checked into source. You probably forgot to copy the hidden .gitignore file generated by sam init when you copied files out of sam-app. You have to explicitly copy that file out since it's a linux hidden file (starts with .). That will prevent git from trying to add those files to the index.

Also, python convention is to try to minimize underscores in module names so call this directory metricpublisher.

ylynn reviewed (changes requested):
Change directory name metric_publisher to metricpublisher. Also add skeleton unit test files.

Commit:
Add the .gitignore file and change metric_publisher to metricpublisher
I also deleted the sam-app. I meant to on the first pull request but for some reason it didn't go through properly. All the code in the sam-app is now in metricpublisher, but possibly under different names.

Commit:
Add the .gitkeep file to the schemas directory.

jlhood reviewed (commented):
+1 to Wujing's comment to add skeleton unit test as well.

Commit:
Add unit test structure.
I thought these were already included in the last commit, but they got lost somehow. This is the basic structure for the unit tests.
